Fix dynamic imports

phiresky opened the pull request:
also includes #66

nadeesha commented:
Thanks very much for this. Could you add an integration test with the success and a failure case as well please?

phiresky commented:
added. also replaced npm link with yarn link in the integration script since npm link doesn't work without root (in my case). let me know if i should revert that
